src/main/java/hudson/plugins/git/UserMergeOptions.java

```diff
@@ -19,28 +20,60 @@
 
     private String mergeRemote;
     private String mergeTarget;
-    private String mergeStrategy;
+    private MergeCommand.Strategy mergeStrategy;
     private MergeCommand.GitPluginFastForwardMode fastForwardMode;
 
     /**
      * @deprecated use the new constructor that allows to set the fast forward mode.
      */
     @Deprecated
     public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy) {
-        this(mergeRemote, mergeTarget, mergeStrategy, MergeCommand.GitPluginFastForwardMode.FF);
+        this(mergeTarget);
+        setMergeRemote(mergeRemote);
+        setMergeStrategy(mergeStrategy);
+        setFastForwardMode(MergeCommand.GitPluginFastForwardMode.FF);
+    }
+
+    /**
+     * @Deprecated use the new @DataBoundConstructor where you only need to supply the necessary information.
```

Comment:
> it's just not clear why this one exists at all

I was not sure what to do with the constructors. I marked the old constructor with `@Deprecated` in case compatibility at the Java level needed to be preserved, and I created a new one with `@DataBoundConstructor` because that is what fixes the bug. Should I put this explanation in the comment? What would you like me to do here?